File: Java/Tests/GenerateTimes.java
import java.io.File;
import java.io.IOException;
import java.math.BigInteger;
import java.util.Random;

public class GenerateTimes {
    static final int NUM_DATA_POINTS = 1000;
    public static void main(String[] args) {

        // ElGamal
        Random randomGen = new Random();
        for (int i = 0; i < NUM_DATA_POINTS; ++i) {
            ElGamal elGamal = new ElGamal();
            BigInteger e_plainText = BigInteger.valueOf(i + 3);
            int randomKey = randomGen.nextInt();

            try {
                File output_encrypt = new File("ElGamal_encryptTimes.txt");
                Pair<BigInteger> e_encryptedText = elGamal.timedEncrypt(e_plainText, BigInteger.valueOf(randomKey), output_encrypt, ',');

                File output_decrypt = new File("ElGamal_decryptTimes.txt");
                BigInteger e_decryptedText = elGamal.timedDecrypt(e_encryptedText, output_decrypt, ',');
            } catch (IOException e) {
                e.printStackTrace();
            }
        }

        // RSA
        for (int i = 0; i < NUM_DATA_POINTS; ++i) {
            RSA rsa = new RSA();
            BigInteger r_plainText = BigInteger.valueOf(i + 3);

            try {
                File output_encrypt = new File("RSA_encryptTimes.txt");
                BigInteger r_encryptedText = rsa.timedEncrypt(r_plainText, output_encrypt, ',');

                File output_decrypt = new File("RSA_decryptTimes.txt");
                BigInteger r_decryptedText = rsa.timedDecrypt(r_encryptedText, output_decrypt, ',');
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
    }
}

File: C++/Tests/GenerateTimes.cpp
#include "ElGamal.h"
#include "RSA.h"
#include "grabbingRandomPrime.h"

#include <limits>

int main() {
    srand(static_cast<int>(time(NULL)));
    
    const int NUM_DATA_POINTS = 1000;
    
    std::ofstream output;

    // ElGamal
    grabbingRandomPrime randomGen = grabbingRandomPrime();
    for (int i = 0; i < NUM_DATA_POINTS; ++i) {
        ElGamal elGamal = ElGamal();
        long long e_plainText = i + 3;
        int randomKey = randomGen.genRandValue(3, INT_MAX);
        
        output.open("ElGamal_encryptTimes.txt", std::ios_base::app);
        Pair<long long> e_encryptedText = elGamal.timedEncrypt(e_plainText, randomKey, output, ',');
        output.close();
        
        output.open("ElGamal_decryptTimes.txt", std::ios_base::app);
        long long e_decryptedText = elGamal.timedDecrypt(e_encryptedText, output, ',');
        output.close();
    }

    // RSA
    for (int i = 0; i < NUM_DATA_POINTS; ++i) {
        RSA rsa = RSA();
        long long r_plainText = i + 3;
        
        output.open("RSA_encryptTimes.txt", std::ios_base::app);
        long long r_encryptedText = rsa.timedEncrypt(r_plainText, output, ',');
        output.close();

        output.open("RSA_decryptTimes.txt", std::ios_base::app);
        long long r_decryptedText = rsa.timedDecrypt(r_encryptedText, output, ',');
        output.close();
    }
    
    return 0;
}
